### PR TITLE
Improve add repository dialog

### PR DESCRIPTION
**Pull request by madoar**
This PR improves the implementation of our dialog for the addition of a new repository.

### CODE REVIEW

File: phoenicis-javafx/src/main/java/org/phoenicis/javafx/views/mainwindow/settings/addrepository/RepositoryType.java [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+        }
+    },
+
+    /**
+     * {@link org.phoenicis.repository.location.ClasspathRepositoryLocation} repository type
+     */
+    CLASSPATH {
+        @Override
+        public String toString() {
+            return tr("Classpath repository");
```
Comment (qparis) [inline phoenicis-javafx/src/main/java/org/phoenicis/javafx/views/mainwindow/settings/addrepository/RepositoryType.java]: I think that toString should never be translated

File: phoenicis-javafx/src/main/java/org/phoenicis/javafx/views/mainwindow/settings/addrepository/RepositoryType.java [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+            return new ClasspathRepositoryDetailsPanel();
+        }
+    };
+
+    /**
+     * Returns the translated name of the repository type
+     *
+     * @return The translated name of the repository type
+     */
+    public abstract String toString();
```
Comment (qparis) [inline phoenicis-javafx/src/main/java/org/phoenicis/javafx/views/mainwindow/settings/addrepository/RepositoryType.java]: I think you should choose another method name. It is confusing

Comment (madoar) [inline phoenicis-javafx/src/main/java/org/phoenicis/javafx/views/mainwindow/settings/addrepository/RepositoryType.java]: The "problem" here is, that these enums are directly shown inside the `ComboBox`.
The ComboBox automatically uses the result of the `toString` method as the label for the object, therefore if we want to use another method name we would need to write a converter for it, do we want that?

Comment (qparis) [inline phoenicis-javafx/src/main/java/org/phoenicis/javafx/views/mainwindow/settings/addrepository/RepositoryType.java]: Then, you could define a separate abstract method and use it in the definition of toString in the parent class. 
I'm a bit surprised that ComboBox does not offer any way of not using toString thought

**Comment by madoar** [comment]
Is it better like this?

**Pull request merged**